eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -0,0 +1,173 @@
+parameters:
+  - name: Nightly
+    type: boolean
+    default: true
+  - name: ServiceDirectory
+    type: string
+    default: ""
+  - name: Artifacts
+    type: object
+    default: []
+
+jobs:
+  - job: smoke_test_eligibility
+    displayName: Check Smoke Test Eligibility
+    steps:
+      - ${{ if and(eq(parameters.Nightly, false), ne(variables['Skip.Release'], 'true')) }}:
+        - pwsh: |
+            if (!"${{ parameters.ServiceDirectory }}") {
+              Write-Error "ServiceDirectory parameter must be specified"
+              exit 1
+            }
+            $samples = (Get-ChildItem -Path "./sdk/${{ parameters.ServiceDirectory }}/*/samples/javascript/" -Directory
```

Comment:
Just a note, at the moment the typescript and javascript samples are both hand-authored so it might be useful to test them both at some point, not just the JS part.

Reply:
Is now the right time to also recursively find samples? Or will this bit of powershell do that?

Reply:
re: typescript+javascript, I believe the process actually transpiles the typescript and runs both that and the javascript samples. It just reads the javascript directory to extract information from the package.json

re: recursion, at this point I'm only extracting package.json information, and service directories that include samples. This detection code actually runs again during the smoke test scripts. I had to copy it over here because we don't want to kick off smoke tests for pipelines where none of the packages being released have any smoke test samples themselves for their respective service (i.e. they have a skip configuration in the package.json).

Reply:
> re: recursion, at this point I'm only extracting package.json information, and service directories that include samples. This detection code actually runs again during the smoke test scripts. I had to copy it over here because we don't want to kick off smoke tests for pipelines where none of the packages being released have any smoke test samples themselves for their respective service (i.e. they have a skip configuration in the package.json).

I can't speak to the entire process (there's only so much powershell I can read!) but I did note that the smoke test runs never ran scripts in the 'advanced' subfolder in service bus, for instance. I'm not sure why it doesn't do that but I assumed this samples discovery bit was responsible.

(for instance, this one would have definitely caused issues in the smoke tests because it runs forever):
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/samples/javascript/advanced/sessionRoundRobin.js

Reply:
Oh, I misunderstood your question! Yeah, I can add that in.